File: demos/jans-chip/ios/Jans-Chip/Database/DBModel/OIDCClient.swift
//
//  OIDCClient.swift
//  Jans-Chip
//
//  Created by Nazar Yavornytskyi on 09.10.2023.
//

import RealmSwift

final class OIDCClient: Object, ObjectKeyIdentifiable, ErrorHandler {
    var errorMessage: String?
    var isSuccess: Bool = true
    
    override static func primaryKey() -> String? {
        "id"
    }
    
    @Persisted var id: String = "1"
    @Persisted var sno: String
    @Persisted var clientName: String
    @Persisted var clientId: String
    @Persisted var clientSecret: String
    @Persisted var recentGeneratedIdToken: String
    @Persisted var recentGeneratedAccessToken: String
    @Persisted var scope: String
}

File: demos/jans-chip/android/app/src/main/java/io/jans/chip/model/OIDCClient.kt
package io.jans.chip.model

import androidx.annotation.NonNull
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import com.google.gson.annotations.SerializedName

@Entity(tableName = "OIDC_CLIENT")
data class OIDCClient(
    @NonNull
    @PrimaryKey(autoGenerate = false)
    @SerializedName("SNO")
    var sno: String,

    @ColumnInfo(name = "CLIENT_NAME")
    @SerializedName("CLIENT_NAME")
    var clientName: String?,

    @ColumnInfo(name = "CLIENT_ID")
    @SerializedName("CLIENT_ID")
    var clientId: String?,

    @ColumnInfo(name = "CLIENT_SECRET")
    @SerializedName("CLIENT_SECRET")
    var clientSecret: String?,

    @ColumnInfo(name = "RECENT_GENERATED_ID_TOKEN")
    @SerializedName("RECENT_GENERATED_ID_TOKEN")
    var recentGeneratedIdToken: String?,

    @ColumnInfo(name = "RECENT_GENERATED_ACCESS_TOKEN")
    @SerializedName("RECENT_GENERATED_ACCESS_TOKEN")
    var recentGeneratedAccessToken: String?,

    @SerializedName("scope")
    var scope: String?,


) {
    @Ignore
    var isSuccessful: Boolean = false

    @Ignore
    var errorMessage: String = ""
}